lib/ui/painting.dart

```diff
@@ -144,35 +168,66 @@ class Color {
   /// * Bits 16-23 are the red value.
   /// * Bits 8-15 are the green value.
   /// * Bits 0-7 are the blue value.
-  final int value;
+  @Deprecated('Use component accessors like .r or .g.')
+  int get value =>
+      ((a * 255.0).round() << 24) |
+      ((r * 255.0).round() << 16) |
+      ((g * 255.0).round() << 8) |
+      ((b * 255.0).round() << 0);
 
   /// The alpha channel of this color in an 8 bit value.
   ///
   /// A value of 0 means this color is fully transparent. A value of 255 means
   /// this color is fully opaque.
-  int get alpha => (0xff000000 & value) >> 24;
+  @Deprecated('Use .a.')
+  int get alpha => (a * 255.0).round();
 
   /// The alpha channel of this color as a double.
   ///
   /// A value of 0.0 means this color is fully transparent. A value of 1.0 means
   /// this color is fully opaque.
-  double get opacity => alpha / 0xFF;
+  @Deprecated('Use .a.')
+  double get opacity => a;
 
   /// The red channel of this color in an 8 bit value.
-  int get red => (0x00ff0000 & value) >> 16;
+  @Deprecated('Use .r.')
+  int get red => (r * 255.0).round();
 
   /// The green channel of this color in an 8 bit value.
-  int get green => (0x0000ff00 & value) >> 8;
+  @Deprecated('Use .g.')
+  int get green => (g * 255.0).round();
 
   /// The blue channel of this color in an 8 bit value.
-  int get blue => (0x000000ff & value) >> 0;
+  @Deprecated('Use .b.')
+  int get blue => (b * 255.0).round();
+
+  Color change(
```

Comment:
The design doc had this written as "with()" but dart doesn't allow that name.  ( https://github.com/dart-lang/language/issues/4020 )

Reply:
I don't love `change`. Maybe `withValues`?

Reply:
`withValues` sounds good to me.  I misremembered thinking that `change` was used elsewhere, hoping to latch onto previous places where we couldn't use `with`.